framework/src/meshmodifiers/AssignElementSubdomainID.C

```diff
@@ -0,0 +1,52 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "AssignElementSubdomainID.h"
+
+template<>
+InputParameters validParams<AssignElementSubdomainID>()
+{
+  InputParameters params = validParams<MeshModifier>();
+  params.addRequiredParam<std::vector<SubdomainID> >("subdomain_ids", "New subdomain IDs of all elements");
+  return params;
+}
+
+AssignElementSubdomainID::AssignElementSubdomainID(const InputParameters & parameters) :
+    MeshModifier(parameters)
+{
+}
+
+AssignElementSubdomainID::~AssignElementSubdomainID()
+{
+}
+
+void
+AssignElementSubdomainID::modify()
+{
+  // Check that we have access to the mesh
+  if (!_mesh_ptr)
+    mooseError("_mesh_ptr must be initialized before calling SubdomainBoundingBox::modify()");
+
+  // Reference the the libMesh::MeshBase
+  MeshBase & mesh = _mesh_ptr->getMesh();
+
+  std::vector<SubdomainID> bids = getParam<std::vector<SubdomainID> >("subdomain_ids");
+
+  if (bids.size() != mesh.n_elem())
+    mooseError(" Size of subdomain_ids in not consistent with the number of elements");
+
+  for (unsigned int e=0; e<mesh.n_elem(); e++)
+    mesh.elem(e)->subdomain_id() = bids[e];
```

Comment:
If this isn't going to work with general meshes... then I would vote for putting this thing in Yak.  At this point it's just too specific to be in MOOSE...